docs: 7702 and MAv2 staging release (#1355)
alchemyplatform/aa-sdk: site/pages/react/using-7702.mdx

---
title: Using EIP-7702
description: Learn how to use EIP-7702 to enable smart account features for EOAs
---

# Using EIP-7702

## What is EIP-7702?

> [EIP-7702](https://eips.ethereum.org/EIPS/eip-7702) is an upcoming Ethereum upgrade set to launch in the coming months as part of the Pectra hardfork. Specifically, EIP-7702 enables Externally Owned Accounts (EOAs) to use smart contract account features. This upgrade marks a pivotal moment in Ethereum’s journey toward account abstraction, a breakthrough designed to make onchain experiences smoother, smarter, and more accessible. EIP-7702 doesn’t replace Ethereum’s existing AA standard, ERC-4337. EIP-7702 was designed to be complementary to ERC-4337 and make it easier for those account abstraction features to reach users.

Reviewer: ```suggestion
[EIP-7702](https://eips.ethereum.org/EIPS/eip-7702) is an upcoming upgrade set to launch as part of the Ethereum Pectra hardfork. Specifically, EIP-7702 enables Externally Owned Accounts (EOAs) to use smart contract account features.

This upgrade marks a pivotal moment in Ethereum’s journey toward account abstraction, a breakthrough designed to make onchain experiences smoother, smarter, and more accessible. EIP-7702 doesn’t replace Ethereum’s existing AA standard, ERC-4337. EIP-7702 was designed to be complementary to ERC-4337 and make it easier for those account abstraction features to reach users.
```

Reviewer: For spacing/readability, and for us to not have to go back and edit the "in the coming months" part after we pass that milestone

- EIP-7702 enables EOAs to delegate control to smart contract accounts that can execute code directly from their addresses
- It is compatible with the prevailing AA standard, ERC-4437, so existing EOAs can enable smart account features without creating a new address and transferring assets
- For users, it will enable web2-like features, including account recovery and passkey sign-in for easier authentication, and web3 benefits like gas-sponsored transaction fees and paying for gas in any token, not just ETH

[Learn more about EIP-7702](https://www.alchemy.com/blog/eip-7702-ethereum-pectra-hardfork).

## When to use EIP-7702?

**For developers with existing users and embedded wallets**

> Upgrade your existing embedded EOA users to smart accounts to enable features like batching and gas sponsorship. Account Kit makes this upgrade easy with compatible smart accounts and infrastructure. Follow [this guide](#TODO/smart-contracts/modular-account-v2/using-7702).

Reviewer: Flagging that this guide does not exist atm

**For developers who have not deployed their app**

> While ERC-4337 and EIP-7702 both enable account abstraction, they serve different needs in modern blockchain applications. You can either:

Reviewer: ```suggestion
While ERC-4337 and EIP-7702 both enable account abstraction, they serve different needs. You can either:
```

> - (Recommended) Use ERC-4337 smart accounts from the start to leverage the fullest of Ethereum AA features such as maximum account programability, extensive transaction customization, and established tooling: get started [here](/react/how-to-set-up-smart-account-client)

Reviewer: ```suggestion
- **(Recommended)** Use ERC-4337 smart accounts from the start to leverage the fullest of Ethereum AA features such as maximum account programability, extensive transaction customization, and established tooling: get started [here](/react/how-to-set-up-smart-account-client)
```

- Use EIP-7702 smart EOAs: keep reading this guide

> See our full [recommendations](https://www.alchemy.com/blog/eip-7702-ethereum-pectra-hardfork#for-developers-that-havent-yet-deployed-their-application) to decide what's best for you.

Reviewer: ```suggestion
See our full [recommendations](https://www.alchemy.com/blog/eip-7702-ethereum-pectra-hardfork#adoption-and-security-considerations-for-application-developers) to decide what's best for you.
```

## How to ship smart EOAs

> In this example, we will upgrade a **new** embedded EOA to a smart contract account ([Modular Account v2](/smart-contracts/modular-account-v2/overview) specifically) and send a gasless transaction from the account.

Reviewer: ```suggestion
In this example, we will upgrade a **new** embedded EOA to a smart contract account ([Modular Account v2](/smart-contracts/modular-account-v2/overview)) and send a gasless transaction from the account.
```
